mk/config.mk

```diff
@@ -725,3 +725,25 @@ CFG_DRIVERS_CLK_FIXED ?= $(CFG_DRIVERS_CLK_DT)
 
 $(eval $(call cfg-depends-all,CFG_DRIVERS_CLK_DT,CFG_DRIVERS_CLK CFG_DT))
 $(eval $(call cfg-depends-all,CFG_DRIVERS_CLK_FIXED,CFG_DRIVERS_CLK_DT))
+
+# Branch Target Identification (part of the ARMv8.5 Extensions) provides a
+# mechanism to limit the set of locations to which computed branch instructions
+# such as BR or BLR can jump. To make use of BTI in TEE cores and ldelf on CPU's
+# that support it, enable this option.
+CFG_CORE_BTI ?= n
+
+$(eval $(call cfg-depends-all,CFG_CORE_BTI,CFG_ARM64_core))
+
+# To make use of BTI in user space libraries and TA's on CPU's that support it,
+# enable this option
+CFG_TA_BTI ?= n
+
+$(eval $(call cfg-depends-all,CFG_TA_BTI,CFG_ARM64_core))
+
+ifeq (y-y,$(CFG_VIRTUALIZATION)-$(call cfg-one-enabled, CFG_TA_BTI CFG_CORE_BTI))
+$(error CFG_VIRTUALIZATION and BTI are currently incompatible)
+endif
+
+ifeq (y-y,$(CFG_WITH_PAGER)-$(call cfg-one-enabled, CFG_TA_BTI CFG_CORE_BTI))
```

Comment:
Shouldn't this be more fine grained? I mean `CFG_WITH_PAGER` and `CFG_CORE_BTI` being mutually exclusive for now is OK, but for `CFG_TA_BTI` shouldn't we check `CFG_PAGED_USER_TA` instead?

Reply:
Thanks  for noticing this. Actually I realized that the issue is only with CFG_PAGED_USER_TA and CFG_TA_BTI since we try to remap the mapped entries again if bti is enabled in the TA elf. I will granularize it further.